be/src/runtime/stream_load/stream_load_pipe.h

```diff
@@ -99,23 +105,13 @@ class StreamLoadPipe : public MessageBodySink {
         // finished
         if (_buf_queue.empty()) {
             DCHECK(_finished);
-            *buf_sz = 0;
-            return Status::OK();
+            return Status::EndOfFile("all data has been read");
         }
-        auto raw = _buf_queue.front();
-        auto raw_sz = raw->remaining();
-
-        if (*buf_cap < raw_sz + padding) {
-            buf->reset(new uint8_t[raw_sz + padding]);
-            *buf_cap = raw_sz + padding;
-        }
-        raw->get_bytes((char*)(buf->get()), raw_sz);
-        *buf_sz = raw_sz;
-
+        auto buf = _buf_queue.front();
```

Comment:
```suggestion
        auto buf = std::move(_buf_queue.front());
```

Reply:
The `std::deque::front` returns a reference to the first element in the container. I think this change may be unnecessary.

Reply:
You are using `auto` not `auto&` here, by default it's a copy.

Reply:
Got that. Updated as `auto& buf = _buf_queue.front();`